src/leases/components/leaseSections/summary/LeaseHistoryItem.js

```diff
@@ -22,24 +22,55 @@ import type {UsersPermissions as UsersPermissionsType} from '$src/usersPermissio
 
 type Props = {
   active?: boolean,
-  id?: number,
   indented?: boolean,
-  lease: Lease,
+  lease?: Lease,
+  id?: number,
+  deleteId?: number,
+  itemTitle?: string,
+  plotSearchName?: string,
+  startDate?: string,
+  endDate?: string,
+  receivedAt?: string,
+  plotSearchType?: string,
+  plotSearchSubtype?: string,
+  applicantName?: string,
+  itemType?: string,
   onDelete?: Function,
   stateOptions: Array<Object>,
   usersPermissions: UsersPermissionsType,
 }
 
-const RelatedLeaseItem = ({
-  active = false,
-  id,
-  indented = false,
-  lease,
-  onDelete,
-  stateOptions,
-  usersPermissions,
-}: Props) => {
-  const identifier = getContentLeaseIdentifier(lease);
+const LeaseHistoryItem = (
+  {
+    active = false,
+    indented = false,
+    lease,
+    id,
+    deleteId,
+    itemTitle = '',
+    startDate,
+    endDate,
+    receivedAt,
+    plotSearchType,
+    plotSearchSubtype,
+    applicantName,
+    itemType = '',
+    onDelete,
+    stateOptions,
+    usersPermissions
+  }: Props,
+) => {
+  const titleString = lease ? getContentLeaseIdentifier(lease) : itemTitle;
+  const MAX_TITLE_LENGTH = 16
+  const title = getTitleText(titleString, MAX_TITLE_LENGTH)
+  const externalLinkHref = lease ? `${getRouteById(Routes.LEASES)}/${lease.id}`
+    : itemType === 'Haku' && id ? `${getRouteById(Routes.PLOT_SEARCH)}/${id}`
+    : itemType === 'Hakemus' && id ? `${getRouteById(Routes.PLOT_APPLICATIONS)}/${id}`
+    : itemType === 'Aluehakemus' && id ? `${getRouteById(Routes.AREA_SEARCH)}/${id}`
+    : null
+  // TODO: Add permissions for deleting plot searches and plot applications
```

Comment:
What are these TODO's here?

Reply:
Actually, what needs to be done is to rename permissions `delete_relatedlease` and `add_relatedlease`.

Reply:
Will you be able to do that and remove the TODO's?